src/packages/migrate/src/Migrate.ts

```diff
@@ -268,6 +266,62 @@ export class Migrate {
     }
   }
 
+  public async tryToRunGenerate(): Promise<void> {
+    const message: string[] = []
+
+    console.info() // empty line
+    logUpdate(
+      `Running generate... ${chalk.dim(
+        '(Use --skip-generate to skip the generators)',
+      )}`,
+    )
+
+    try {
+      const generators = await getGenerators({
+        schemaPath: this.schemaPath,
+        printDownloadProgress: false,
+        version: packageJson.prisma.version,
+        cliVersion: packageJson.version,
+      })
+
+      for (const generator of generators) {
+        const toStr = generator.options!.generator.output!
+          ? chalk.dim(
+              ` to .${path.sep}${path.relative(
+                process.cwd(),
+                generator.options!.generator.output!,
+              )}`,
+            )
+          : ''
+        const name = generator.manifest
+          ? generator.manifest.prettyName
+          : generator.options!.generator.provider
+
+        logUpdate(`Running generate... - ${name}`)
+
+        const before = Date.now()
+        try {
+          await generator.generate()
+          const after = Date.now()
+          const version = generator.manifest?.version
+          message.push(
+            `✔ Generated ${chalk.bold(name!)}${
+              version ? ` (version: ${version})` : ''
+            }${toStr} in ${formatms(after - before)}`,
+          )
+          generator.stop()
+        } catch (err) {
+          message.push(`${err.message}`)
+          generator.stop()
+        }
+      }
+    } catch (errGetGenerators) {
```

Comment:
Why do you `catch` and then directly `throw` it again?

Reply:
I guess because it's a copy paste from another place where we where doing more than throwing 😅

Reply:
Ok, you can also merge and clean up after

Reply:
Sounds good to me!